pkg/cmd/pulumi/auth/login.go

```diff
@@ -205,19 +212,29 @@ func NewLoginCmd() *cobra.Command {
 	return cmd
 }
 
-func validateCloudBackendType(typ string) error {
+func validateCloudBackendType(typ string) (bool, error) {
 	kind := strings.SplitN(typ, ":", 2)[0]
 	supportedKinds := []string{"azblob", "gs", "s3", "file", "https", "http"}
 	for _, supportedKind := range supportedKinds {
 		if kind == supportedKind {
-			return nil
+			if strings.Contains(kind, "http") {
+				return true, nil
+			}
+			return false, nil
 		}
 	}
-	return fmt.Errorf("unknown backend cloudUrl format '%s' (supported Url formats are: "+
+	return false, fmt.Errorf("unknown backend cloudUrl format '%s' (supported Url formats are: "+
 		"azblob://, gs://, s3://, file://, https:// and http://)",
 		kind)
 }
 
+func checkHTTPCloudBackenUrl(url string) (string, error) {
+	if strings.HasSuffix(strings.TrimSuffix(url, "/"), "pulumi.com") {
```

Comment:
```suggestion
	if strings.HasSuffix(strings.TrimSuffix(url, "/"), ".pulumi.com") {
```

I think this is needed to avoid breaking e.g. `pulumi login mycorp-pulumi.com`.  Unlikely someone has this, but still better to avoid the chance of breaking it.

Would be nice to have a test for this as well.